### ISSUE TITLE
Please backport #13321 to `1.9.x` branch

### ISSUE THREAD

**Issue by jeffwidman** [hashicorp/vault#13528]
Please backport #13321 to the `1.9.x` branch. 

Currently if we want to pull in vault, we have to manually add the follow replace directives to workaround the brokenness:
```
replace (
	github.com/hashicorp/vault/api/auth/approle => github.com/hashicorp/vault/api/auth/approle v0.1.0
	github.com/hashicorp/vault/api/auth/userpass => github.com/hashicorp/vault/api/auth/userpass v0.1.0
)
```

Already fixed on `main` via #13321, so it just needs to get backported.

**Comment by digivava** [comment]
Thanks for noticing this. Just backported it to 1.9.x.

**Comment by jeffwidman** [comment]
Thanks.

Also, don't forget in your PR fixing this to say something like `Fix #13528` so that it auto closes / auto-links to this for those who want to check how it was fixed.

Looks like it was landed in https://github.com/hashicorp/vault/pull/13580
